packages/bot-skeleton/src/utils/strategy-helper.js

```diff
@@ -0,0 +1,72 @@
+/** Helper function to perform sequential execution of functions */
+export const pipe = (...fns) => {
+    return initialValue => {
+        return fns.reduce((nextParam, fn) => fn(nextParam), initialValue);
+    };
+};
+
+/** Helper function to parse block information from XML */
+export const extractBlocksFromXml = xml => {
+    const parser = new DOMParser();
+    const xmlDoc = parser.parseFromString(xml, 'text/xml');
+    const blocks = xmlDoc.getElementsByTagName('block');
+
+    const blocksTypeAndFields = Array.from(blocks).map(block => {
+        const type = block.getAttribute('type');
+        const fields = Array.from(block.getElementsByTagName('field')).map(field => {
+            return {
+                name: field.getAttribute('name'),
+                value: field.textContent.trim(),
+            };
+        });
+        return { type, fields };
+    });
+
+    return blocksTypeAndFields;
+};
+
+/** Helper function to sort block childs based on type or field name */
+export const sortBlockChild = blocksArray =>
+    blocksArray.sort((a, b) => {
+        if (a.type < b.type) {
+            return -1;
+        }
+        if (a.type > b.type) {
+            return 1;
+        }
+        return JSON.stringify(a.fields) < JSON.stringify(b.fields) ? -1 : 1;
+    });
+
+/** Compare two blocks by field name and value */
+const areFieldsEqual = (fields1, fields2) => {
+    if (fields1?.length !== fields2?.length) {
+        return false;
+    }
+    for (let i = 0; i < fields1.length; i++) {
+        if (fields1[i]?.name !== fields2[i]?.name || fields1[i]?.value !== fields2[i]?.value) {
+            return false;
+        }
+    }
+    return true;
```

Comment:
@shafin-deriv maybe we can use the `.every()` method for an array of objects in a more modern way:

const fields1 = [
    {'value':'a', 'name': 'a'},
    {'value':'b', 'name': 'b'},
    {'value':'c', 'name': 'c'}
    ];
const fields2 = [
    {'value':'a', 'name': 'a'},
    {'value':'b', 'name': 'b'},
    {'value':'c', 'name': 'c'}
    ];
    
`let isEqual = fields1.every((value, index) => value.name === fields2[index].name || value.value === fields2[index].value);`

same with other `for` loops

Reply:
@maryia-matskevich-deriv that would be a really good approach! But for these comparisons can we stick to the for loop? In this case, the calculation is really heavy and as you know for loop is more performant than the array-based methods (just for reference: https://blog.bitsrc.io/finding-the-fastest-loop-type-in-javascript-38af16fe7b4f) so wouldn't it be better if we stick to the for loops only for these 2 cases? I also just ran a benchmark between the 2 approaches in the real code for Martingale the differences were exactly the same but for a custom strategy which is a bit longer strategy, the results were [every: 0.10000000149011612 and for loop: 0.09999999403953552]. Though it's not that much of a difference but maybe for longer/bigger strategies may be the differences could be higher. Please let me know if you want to have a discussion regarding this.

Reply:
@shafin-deriv 
you have to realize they're human beings and they can make mistakes. From more sources of information, we can conclude that the cycle `for` is indeed faster, but this speed is comparatively negligible. Besides, it depends on the browser. But of course, you should move forward from the situation. 
Two years ago, my guys from the Belarusian office and I organized a demolition of this theory on this or a similar article, with graphs of speed results. After our testing locally, we got different statistics. After that, I began to distrust such reports. 
In general, I tend to this thought: "In conclusion, for loops may be faster depending on the browser being used but array methods should still be used wherever possible because of their readability". ( excerpt from the articlehttps://medium.com/@gabriellegianna92/loops-vs-array-methods-26999051ba45)
the choice of writing style is yours 😉 
it's important that our code be the best👌

Reply:
Thanks for the explanation @maryia-matskevich-deriv! If it is okay then let us stick to the for loop for this case.

Reply:
One thing which I think we should consider is , in case of complex and big strategy and the algorithm is already at O(n^2) time complexity. so let's stick to whichever is faster, what you guys say ?